From 86f47d369d99f0eabd968c0eed9dca94d9c0648b Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 12 Jul 2024 10:36:56 +0200
Subject: [PATCH 1188/1695] plat-stm32mp1: pager use SYSRAM last page if
 possible

Update stm32mp1 with pager TZSRAM size to use SYSRAM last page
now that pager implementation issue pager pageable boundary is
addressed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 227c552bdedc554a1c437c73ce1d63f4522a1d16 plat-stm32mp1: pager use SYSRAM last page if possible]
Change-Id: If80d9e10edeb9a3cffd310c29a161519e07286bb
[etienne: backport from upstream 932059bfed12a8024a995894a1c1c7ef86e9c1c1]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403920
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index cb2911a59..5ad6bb6e6 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -284,19 +284,6 @@ CFG_TUI_FRAME_BUFFER_SIZE_MAX ?= 0x01000000
 CFG_RESERVED_VASPACE_SIZE ?= (10 * 1024 * 1024 + $(CFG_TUI_FRAME_BUFFER_SIZE_MAX))
 endif
 
-ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-512M)),)
-CFG_DRAM_SIZE    ?= 0x20000000
-endif
-
-CFG_DRAM_BASE    ?= 0xc0000000
-CFG_DRAM_SIZE    ?= 0x40000000
-ifeq ($(CFG_STM32MP15)-$(CFG_WITH_PAGER),y-y)
-CFG_TZSRAM_START ?= 0x2ffc0000
-CFG_TZSRAM_SIZE  ?= 0x0003f000
-endif #CFG_STM32MP15 && CFG_WITH_PAGER
-CFG_TZDRAM_SIZE  ?= 0x02000000
-CFG_TZDRAM_START ?= ($(CFG_DRAM_BASE) + $(CFG_DRAM_SIZE) - $(CFG_TZDRAM_SIZE))
-
 # CFG_STM32MP1_SCMI_SHM_BASE and CFG_STM32MP1_SCMI_SHM_SIZE define the
 # device memory mapped SRAM used for SCMI message transfers.
 # When CFG_STM32MP1_SCMI_SHM_BASE is set to 0, the platform uses OP-TEE
@@ -308,11 +295,25 @@ CFG_TZDRAM_START ?= ($(CFG_DRAM_BASE) + $(CFG_DRAM_SIZE) - $(CFG_TZDRAM_SIZE))
 CFG_STM32MP1_SCMI_SHM_SYSRAM ?= n
 ifeq ($(CFG_STM32MP1_SCMI_SHM_SYSRAM),y)
 $(call force,CFG_STM32MP1_SCMI_SHM_BASE,0x2ffff000)
+CFG_TZSRAM_SIZE  ?= 0x0003f000
 else
 CFG_STM32MP1_SCMI_SHM_BASE ?= 0
 endif
 $(call force,CFG_STM32MP1_SCMI_SHM_SIZE,0x1000)
 
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-512M)),)
+CFG_DRAM_SIZE    ?= 0x20000000
+endif
+
+CFG_DRAM_BASE    ?= 0xc0000000
+CFG_DRAM_SIZE    ?= 0x40000000
+ifeq ($(CFG_STM32MP15)-$(CFG_WITH_PAGER),y-y)
+CFG_TZSRAM_START ?= 0x2ffc0000
+CFG_TZSRAM_SIZE  ?= 0x00040000
+endif #CFG_STM32MP15 && CFG_WITH_PAGER
+CFG_TZDRAM_SIZE  ?= 0x02000000
+CFG_TZDRAM_START ?= ($(CFG_DRAM_BASE) + $(CFG_DRAM_SIZE) - $(CFG_TZDRAM_SIZE))
+
 CFG_STM32_BSEC ?= y
 CFG_STM32_CLKCALIB ?=y
 CFG_STM32_CPU_OPP ?= y
-- 
2.47.2

